File: Lab_ProblemSet/Caesar.php
<?php
// シーザー暗号
function main(int $key): void {
    $key %= 26;
    echo 'plaintext: ';
    $word = trim(fgets(STDIN));
    $alphabet = range('a', 'z');
    $ciphertext = '';

    for ($i=0; $i < strlen($word); $i++) {
        if (!ctype_alpha($word[$i])) {
            $ciphertext .= $word[$i];
            continue;
        }

        for ($j = 0; $j < count($alphabet); $j++) {
            if ($alphabet[$j] === $word[$i]) {
                $ciphertext .= $alphabet[$j + $key];
            } elseif (strtoupper($alphabet[$j]) === $word[$i]) {
                $ciphertext .= strtoupper($alphabet[$j + $key]);
            }
        }
    }
    echo 'ciphertext: ', $ciphertext, PHP_EOL;
}

if (count($argv) != 2) return 1;
if ($argv[1] < 1) return 1;

main($argv[1]);
?>

File: Lab_ProblemSet/caesar.py
# Caesar Cipher
import sys


def main(key: int) -> None:
    key %= 26
    word = input("plaintext: ").strip()
    alphabet = [chr(i) for i in range(97, 123)]  # 'a' to 'z'
    ciphertext = ""

    for char in word:
        if not char.isalpha():
            ciphertext += char
            continue

        for i, alpha in enumerate(alphabet):
            if alpha == char:
                ciphertext += alphabet[(i + key) % 26]
            elif alpha.upper() == char:
                ciphertext += alphabet[(i + key) % 26].upper()

    print(f"ciphertext: {ciphertext}")

if __name__ == "__main__":
    if len(sys.argv) != 2:
        sys.exit(1)
    if int(sys.argv[1]) < 1:
        sys.exit(1)

    main(int(sys.argv[1]))
